include/boost/unordered/detail/foa/table.hpp

```diff
@@ -216,8 +236,8 @@ class table_iterator
     serialize_tracked_address(ar,p);
   }
 
-  unsigned char      *pc=nullptr;
-  table_element_type *p=nullptr;
+  char_pointer          pc=nullptr;
+  table_element_pointer  p=nullptr;
 };
```

Comment:
Maybe there's an overabundance of `boost::to_address(pc)` within `table_iterator` (and in a couple of places outside it). Consider renaming the members to `pc_` and `p` and providing `pc()` and `p()` accessors much like in the arrays classes.